src/Compilers/Core/Portable/Operations/Operation.cs

```diff
@@ -286,6 +285,12 @@ internal IOperation SearchParentOperation()
                         }
                     }
 
+                    if (SemanticModel.Root == currentCandidate)
```

Comment:
I believe you could avoid duplicating this check by doing it at the beginning of the loop and setting `currentCandidate` to `Syntax` at the beginning of the loop.

Reply:
> I believe you could avoid duplicating this check by doing it at the beginning of the loop and setting currentCandidate to Syntax at the beginning of the loop.

Then the loop would have to be changed to ```while (true) and the null check moved inside. I believe the condition is simple enough that duplication is not an issue.